dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,225 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    def set_up_state_api(self):
+        """
+        Set up the state API in order to fetch task information.
+        """
+        gcs_channel = self._dashboard_head.aiogrpc_gcs_channel
+        self._state_api_data_source_client = StateDataSourceClient(
+            gcs_channel, self._dashboard_head.gcs_aio_client
+        )
+        self._state_api = StateAPIManager(self._state_api_data_source_client)
+
+    async def get_worker_details_for_task(self, task_id, attempt_number):
+        """
+        Retrieves worker details for a specific task and attempt number.
+
+        Args:
+            task_id (str): The ID of the task.
+            attempt_number (int): The attempt number of the task.
+
+        Returns:
+            tuple: A tuple containing the worker's PID (process ID) and IP address.
+
+        Raises:
+            ValueError: If the PID or IP is None, indicating that the information could not be fetched.
+            ValueError: If the task attempt is not in the "RUNNING" state.
+        """
+        option = ListApiOptions(
+            filters=[
+                ("task_id", "=", task_id),
+                ("attempt_number", "=", attempt_number),
+            ],
+            detail=True,
+            timeout=10,
+        )
+
+        result = await self._state_api.list_tasks(option=option)
+        tasks = result.result
+
+        pid = tasks[0]["worker_pid"]
+        logger.info(f"pid {type(pid)}: {pid}")
+
+        node_id = tasks[0]["node_id"]
+        ip = DataSource.node_id_to_ip[node_id]
+
+        state = tasks[0].get("state")
+        logger.info(f"ip {type(ip)}: {ip}")
+
+        if not pid or not ip:
+            raise ValueError(
+                f"pid or ip is None, could not fetch the info you need: pid = {pid}, ip = {ip}"
+            )
+        logger.info(f"node_id {type(node_id)}: {node_id}")
+
+        if state != "RUNNING":
```

Comment:
should this check be part of this function? Or if so, the name of the function seems to mismatch with this behaviour.

Reply:
I agree, it deny the Single Responsibility Principle

Reply:
I created a new function to validate the errors